### PR TITLE
trans_hugepage_relocated: fix some issues to make it run correctly

### PR DESCRIPTION
**Pull request by yanan-fu**
depends on: https://github.com/avocado-framework/avocado-vt/pull/3335

id: 2032517
Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
On s390x:
(5/5) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.trans_hugepage.relocated.s390-virtio: ERROR: Hugepages allocated can not reach a half: 0/1024

@yanan-fu I've add the log in our bz, please check

**Comment by zhenyzha** [comment]
Very strange, the program seems to sleep after entering the guest and does not continue to run. No error is reported after waiting for more than 4 hours, and the test can only be interrupted manually.

Add detailed log to bug.

**Comment by zhenyzha** [comment]
BTW 
https://github.com/autotest/tp-qemu/blob/master/generic/tests/cfg/trans_hugepage.cfg#L21
The `thp_test_config` path needs to be modified on ARM, I have corrected it locally, but the auto case is still not completed.
on ARM 
`thp_test_config = "/sys/kernel/mm/transparent_hugepage/khugepaged/pages_to_scan:4096;/sys/kernel/mm/transparent_hugepage/khugepaged/scan_sleep_millisecs:10000;/sys/kernel/mm/transparent_hugepage/khugepaged/alloc_sleep_millisecs:60000"`

**Comment by fbq815** [comment]
> On s390x: (5/5) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.trans_hugepage.relocated.s390-virtio: ERROR: Hugepages allocated can not reach a half: 0/1024
> 
> @yanan-fu I've add the log in our bz, please check

@yanan-fu On s390x, VM memory backing will not cost THP in the host, please add no s390x for this case as well

File: generic/tests/trans_hugepage_relocated.py [hunk 1]
```diff
@@ -73,21 +73,20 @@ def nr_hugepage_check(sleep_time, wait_time):
         mem_free_filter = r"MemFree:\s+(.\d+)\s+(\w+)"
         guest_mem_free, guest_unit = re.findall(mem_free_filter, o)[0]
         if re.findall("[kK]", guest_unit):
-            guest_mem_free = str(int(guest_mem_free) / 1024)
+            guest_mem_free = int(guest_mem_free) / 1024
```
Comment (hereischen) [inline generic/tests/trans_hugepage_relocated.py:76]: python 3 you will get a float by doing this
>>> 2000/1024
1.953125

Comment (hereischen) [reply generic/tests/trans_hugepage_relocated.py:76]: use `//`?

Comment (zhenyzha) [reply generic/tests/trans_hugepage_relocated.py:76]: good points :)

**Comment by mcasquer** [comment]
@yanan-fu any update about this? In [my last comment](https://bugzilla.redhat.com/show_bug.cgi?id=2032517#c6) I suggested to change the `vmsm` variable value, is there any other suggestion?

**Pull request closed without merging**